src/s2s/pipeline.cpp

```
@@ -0,0 +1,1756 @@
+    void run(S2SStream& st, int step) { (void)queue_.run(0, Request{&st, step}); }
+    asr::BatchMetrics metrics() const { return queue_.metrics(); }
+
+   private:
+    S2SPipeline* owner_;
+    asr::MicroBatcher<int, Request, uint8_t> queue_;
+};
+
+S2SStream::~S2SStream() {
+    extract_cancelled = true;
+    inject_cancelled = true;
+    if (fast_path_thread.joinable())
```

Clang (14.0.6)

[warning] statement should be inside braces

(readability-braces-around-statements)